Feat: Appreg v2 updates (#6855)
Kong/docs.konghq.com: app/konnect/dev-portal/applications/dynamic-client-registration/azure.md

## Configure the Dev Portal
{% navtabs %}
{% navtab Konnect UI %}
Once you have Azure configured, you can set up the Dev Portal to use Azure for Dynamic Client Registration (DCR). This process involves two steps: creating the DCR provider and establishing the authentication strategy. DCR providers are designed to be reusable configurations. This means once you've configured the Azure DCR provider, it can be used across multiple authentication strategies without needing to be set up again.

1. Log in to {{site.konnect_short_name}} and select the **Dev Portal** {% konnect_icon dev-portal %} from the menu.

2. Go to **Application Auth** to find the authentication settings for your API products.

3. Open the **DCR Providers** to review existing DCR Providers.

4. Select **New DCR Provider** button to create an Azure configuration. Enter a name for internal reference in {{site.konnect_short_name}}. This name, along with the provider type, will remain internal and not be visible to developers on the Dev Portal.

5. Provide the Issuer URL for your Azure tenant, using the format  `https://sts.windows.net/YOUR_TENANT_ID`.

6. For **Provider Type**, select AzureAD.

> 7. Enter your Application (Client) ID from Azure into the **Initial Client ID** field and the Azure admin application's client secret into the **Intial Client Secret** field. The Initial Client Secret will be stored in isolated, encrypted storage and will not be readable through any Konnect API.

Reviewer: ```suggestion
7. Enter your Application (Client) ID from Azure into the **Initial Client ID** field and the Azure admin application's client secret into the **Intial Client Secret** field. The Initial Client Secret will be stored in isolated, encrypted storage and will not be readable through any Konnect API.
```

```sh
   curl --request POST \
   --url https://us.api.konghq.com/v2/dcr-providers \
   --header 'Authorization: $KPAT' \
   --header 'content-type: application/json' \
   --data '{
   "name": "Azure DCR Provider",
   "provider_type": "Azure",
   "issuer": "https://sts.windows.net/YOUR_TENANT_ID",
   "dcr_config": {
      "initial_client_id": "abc123",
      "initial_client_secret": "abc123xyz098!",
      "initial_client_audience": "https://my-custom-domain.com/api/v2/"
   }
   }'
```
You will receive a response that includes a `dcr_provider` object similar to the following:

   ```sh
   {
      "created_at": "2024-02-29T23:38:00.861Z",
      "updated_at": "2024-02-29T23:38:00.861Z",
      "id": "93f8380e-7798-4566-99e3-2edf2b57d289",
      "name": "Azure DCR Provider",
      "provider_type": "Azure",
      "issuer": "https://sts.windows.net/YOUR_TENANT_ID",
      "dcr_config": {
         "initial_client_id": "abc123",
         "initial_client_audience": "https://my-custom-domain.com/api/v2/"
      },
      "active": false
   }

   ```
Save the `id` value for creating the authentication strategy.

> 2. With the `dcr_id` obtained from the first step, create an authentication strategy. Send a `POST` request to the [`create-auth-stratgies`](/konnect/api/application-auth-strategies/latest/#/App%20Auth%20Strategies/create-app-auth-strategy) endpoint describing an authentiaction strategy: 

Reviewer: ```suggestion
2. With the `dcr_id` obtained from the first step, create an authentication strategy. Send a `POST` request to the [`create-auth-stratgies`](/konnect/api/application-auth-strategies/latest/#/App%20Auth%20Strategies/create-app-auth-strategy) endpoint describing an authentication strategy: 
```

5. Provide the Issuer URL for your Azure tenant, using the format  `https://sts.windows.net/YOUR_TENANT_ID`.

6. For **Provider Type**, select AzureAD.

7. Enter your Application (Client) ID from Azure into the **Initial Client ID** field and the Azure admin application's client secret into the **Intial Client Secret** field. The Initial Client Secret will be stored in isolated, encrypted storage and will not be readable through any Konnect API.

8. After saving, your new DCR Provider will appear in the list.

9. Click the **Auth Strategy** tab, then select **New Auth Strategy** to create an auth strategy that uses the DCR Provider you just added.

10. Name it for internal reference in {{site.konnect_short_name}} and provide a display name for visibility in the Portal. Select DCR as the **Auth Type** and choose your newly created DCR provider from the dropdown. The  **Issuer URL** entered earlier will be will be prepopulated.

11. In the **Credential Claims** field, enter `appid`.

12. Select the relevant **Auth Methods** you need (`client_credentials`, bearer, session), and **save**. 

{% endnavtab %}
{% navtab API %}

> After configuring Azure, you can integrate it with the Dev Portal for dynamic client registration (DCR). This process involves two steps: creating the DCR provider and establishing the authentication strategy. DCR providers are designed to be reusable configurations. This means once you've configured the Azure DCR provider, it can be utilized across multiple authentication strategies without needing to be set up again.

Reviewer 2: ```suggestion
After configuring Azure, you can integrate it with the Dev Portal for dynamic client registration (DCR). This process involves two steps: creating the DCR provider and establishing the authentication strategy. DCR providers are designed to be reusable configurations. This means once you've configured the Azure DCR provider, it can be used across multiple authentication strategies without needing to be set up again.
```

> 1. Start by creating the DCR provider. Send a `POST` request to the[`v2/dcr-providers`](https://kong-platform-api.netlify.app/konnect/application-auth-strategies/v2/openapi.yaml/#tag/DCR-Providers/operation/create-dcr-provider) endpoint with your DCR configuration details. 

Reviewer 2: ```suggestion
1. Start by creating the DCR provider. Send a `POST` request to the [`v2/dcr-providers`](https://kong-platform-api.netlify.app/konnect/application-auth-strategies/v2/openapi.yaml/#tag/DCR-Providers/operation/create-dcr-provider) endpoint with your DCR configuration details:
```

      "initial_client_id": "abc123",
      "initial_client_secret": "abc123xyz098!",
      "initial_client_audience": "https://my-custom-domain.com/api/v2/"
   }
   }'
```
You will receive a response that includes a `dcr_provider` object similar to the following:

   ```sh
   {
      "created_at": "2024-02-29T23:38:00.861Z",
      "updated_at": "2024-02-29T23:38:00.861Z",
      "id": "93f8380e-7798-4566-99e3-2edf2b57d289",
      "name": "Azure DCR Provider",
      "provider_type": "Azure",
      "issuer": "https://sts.windows.net/YOUR_TENANT_ID",
      "dcr_config": {
         "initial_client_id": "abc123",
         "initial_client_audience": "https://my-custom-domain.com/api/v2/"
      },
      "active": false
   }

   ```
Save the `id` value for creating the authentication strategy.

2. With the `dcr_id` obtained from the first step, create an authentication strategy. Construct a `POST` request to the [`v2/create-auth-stratgies`](https://kong-platform-api.netlify.app/konnect/application-auth-strategies/v2/openapi.yaml/#tag/App-Auth-Strategies/operation/create-app-auth-strategy) endpoint describing an authentiaction strategy: 

   ```sh
      curl --request POST \

>       --url https://us.api.konghq.tech/v2/application-auth-strategies \

Author: ``
```suggestion
      --url https://us.api.konghq.com/v2/application-auth-strategies \
```

1. Sign in to {{site.konnect_short_name}}, then select {% konnect_icon dev-portal %} **Dev Portal** from the menu.

2. Navigate to **Application Auth** to access the authentication settings for your API Products.

3. Open the **DCR Providers** to view all configured DCR Providers

> 4. Select **New DCR Provider** button to create an Auth0 configuration. Enter a name for internal use in {{site.konnect_short_name}} and a display name for visibility on your portal.

Reviewer 3: ```suggestion
4. Select **New DCR Provider** button to create an Auth0 configuration. Enter a name for internal use in {{site.konnect_short_name}}. The name and provider type information will not be exposed to Dev Portal developers.
```

5. Input the **Issuer URL** of your Azure tenant, formatted as: `https://sts.windows.net/YOUR_TENANT_ID`.

6. Choose AzureAD as the **Provider Type**.

> 7. Enter your Application (Client) ID from Azure in the **Initial Client ID** field. Enter the client secret from the admin application created in Azure into the **Initial Client Secret** field.

Reviewer 3: ```suggestion
7. Enter your Application (Client) ID from Azure in the **Initial Client ID** field. Enter the client secret from the admin application created in Azure into the **Initial Client Secret** field. Note: The Initial Client Secret will be stored in isolated, encrypted storage and will not be readable through any Konnect API.
```
